### ISSUE TITLE
[13.0,14.0] Stock Interim lines are not reconciled for Kits in Anglo-Saxon Accounting

### ISSUE THREAD

**Issue by lem8r** [odoo/odoo#60679]
Impacted versions: 13.0, 14.0

Steps to reproduce:
1. Install CoA with Anglo-Saxon Accounting enabled (generic_coa for example).
2. Configure Product A to be sold as a Kit: phantom BoM containing storable products B, C, D.
3. Sell Product A, confirm the delivery of B, C, D. Create and post Invoice for product A.

Current behavior:
Stock interim outgoing account will have 1 line for product A created from invoice and 3 lines for products B, C, D created from stock move. Total balance is correct but lines are not reconciled together.
The reason is current reconciliation algorithm matches lines by product which is not the case for kits.

Expected behavior:
Stock interim lines should be reconciled for kits as well.
Otherwise there is too much manual work to reconcile them manually. Especially when AMLs from stock move and Invoice have no common reference(SO number).

Video/Screenshot link (optional):
https://github.com/odoo/odoo/blob/14.0/addons/stock_account/models/account_move.py#L173

**Comment by AaronHForgeFlow** [comment]
I confirmed I can reproduce the issue in v13

**Comment by AaronHForgeFlow** [comment]
In purchase flows similar thing happens. See this example:

- One product with BOM type “kit”
- The finished product is consumable. Cost 20 DKK
- The components are stockable. Cost 10 DKK

The stock journal looks like this after receiving:
![image](https://user-images.githubusercontent.com/19620251/112823959-38565280-908a-11eb-9c8c-4dbe7f312bd7.png)

But in the vendor bill the Interim account is not balanced:
![image](https://user-images.githubusercontent.com/19620251/112824052-5754e480-908a-11eb-9f60-ab8c68962758.png)

In case we set the kit as stockable product then similar thing happens, the interim account is not reconciled because the product is not the components.

**Comment by AaronHForgeFlow** [comment]
BTW, just tested in runbot v13 and this is why I get for sales flow:
The conditions are: 
- anglosaxon CoA
- kit of type consumable (that is recommended in the official documentation)  Standard cost 2$
- 2 components storable. Standard cost 1$
-  just create a SO deliver and create the customer invoice

The customer invoice will not create any journal item regarding stock:
![image](https://user-images.githubusercontent.com/19620251/112967226-55068f00-914b-11eb-987e-c44ced5b9e4d.png)

As long as the delivery already debited the Stock Interim Output account, that account is not balanced:
![image](https://user-images.githubusercontent.com/19620251/112967557-af9feb00-914b-11eb-9766-2f467ca68655.png)

Moreover the Expenses account is also not hit.

@lem8r  how did you manage to get the stock related journal items in your example? I could not get that on runbot. Thanks in advance!

**Comment by lem8r** [comment]
@AaronHForgeFlow it should work in your case. Maybe accounts on product's category are not configured.

**Comment by AaronHForgeFlow** [comment]
Thanks for your reply. Actually the accounts seems fine in the product category (same category for the kit and the components)
![image](https://user-images.githubusercontent.com/19620251/112975519-0d383580-9154-11eb-9efc-1e82850b1bce.png)

I will keep checking if I missed anything else. Regards.

**Comment by lem8r** [comment]
Maybe "kit of type consumable"? In my case all products were storable as they can be sold separately or as a kit.

**Comment by AaronHForgeFlow** [comment]
Yes, it has to be that. The official documentation says it has to be consumable if I don't wanna keep stock for the kit product: https://www.odoo.com/documentation/user/14.0/manufacturing/management/kit_shipping.html

however it seems that is not anglo-saxon accounting friendly :)

**Comment by lem8r** [comment]
_Because you cannot track the stock of kit products, the Product Type should be set to Consumable_ comes from v11 documentation. At that time mrp module was not able to compute stock levels of kit products. That feature was introduced in v13.
